### ISSUE TITLE
[Generator] Delegate conversions with generic params are broken

### ISSUE THREAD

**Issue by slxdy** [BepInEx/Il2CppInterop#120]
Idk how to describe it well, so I think the following Il2CppInterop-generated piece speaks for itself.
At `Il2CppSystem.Comparison<>`:
```cs
public static implicit operator Comparison<T>(System.Func<,> P_0)
{
    return DelegateSupport.ConvertDelegate<Comparison<T>>(P_0);
}
```
The parameter type does not specify the generic arguments

**Comment by ds5678** [comment]
What game is this?

**Comment by ds5678** [comment]
For The Long Dark, that method looks fine:
```cs
public static implicit operator Comparison<T>(System.Func<T, T, int> P_0)
{
	return DelegateSupport.ConvertDelegate<Comparison<T>>(P_0);
}
```

**Comment by ds5678** [comment]
I think this was fixed by #132.
